notebooks/MAST/HSC/CASJOBS/HSCV3_SMC_API/hscv3_smc_api.ipynb

```diff
@@ -0,0 +1,516 @@
+    "* `axes` = axis range tuple from fastkde\n",
+    "* `markersize` = matplotlib marker size (which is an area in points)\n",
+    "* `over` = oversampling factor (empirically 5 fills in the holes)\n",
+    "* `verbose` = print additional info\n",
+    "\n",
+    "Returns an array of subscripts to plot from `zs`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def select_subset(zs,axes,markersize=2,over=5,verbose=False):\n",
+    "    \"\"\"Select a subset of points in crowded regions to speed up plotting\n",
+    "    \n",
+    "    Returns array of subscripts to plot from zs    \n",
+    "    \"\"\"\n",
+    "    # get figure size in points\n",
+    "    fig = plt.gcf()\n",
+    "    figsize = fig.get_size_inches()*72.0\n",
+    "    # scale factor to convert PDF values to expected number of points in the marker area\n",
+    "    # divide ranges by 2 because fastkde pads the data ranges to allow FFTs\n",
+    "    range0 = 0.5*(axes[0][-1]-axes[0][0])\n",
+    "    range1 = 0.5*(axes[1][-1]-axes[1][0])\n",
+    "    ss = len(zs) * range0 * range1 * markersize / (over*figsize[0]*figsize[1])\n",
+    "    cc = (1.0/(ss*zs).clip(min=1)).cumsum()\n",
+    "    wsel = np.searchsorted(cc, np.arange(np.ceil(cc[-1]).astype(int)+1),\n",
+    "                           side='right').clip(max=len(zs)-1)\n",
+    "    if verbose:\n",
+    "        print(f\"Plotting {len(wsel)} of {len(zs)} points\")\n",
+    "    return wsel"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Find objects in the SMC <a name=\"smc\"></a>\n",
+    "\n",
+    "This is based on [HSC Use Case #2](https://archive.stsci.edu/hst/hsc/help/use_case_2_v3.html), which includes an example of creating a color-magnitude diagram for the SMC using <a href=\"https://mastweb.stsci.edu/hcasjobs\">MAST CasJobs</a>.  This is simple to do using the HSC API."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Use `astropy` name resolver to get position of the SMC <a name=\"resolver\"></a>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "target = 'SMC'\n",
+    "coord = SkyCoord.from_name(target)\n",
+    "\n",
+    "ra = coord.ra.degree\n",
+    "dec = coord.dec.degree\n",
+    "print(f'ra: {ra}\\ndec: {dec}')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Select objects with the desired magnitudes and colors near the SMC <a name=\"summary\"></a>\n",
+    "\n",
+    "This searches the summary table for objects in a 3x3 degree box centered on the galaxy that have measurements in both ACS F555W and F814W.  It computes the V-I color and selects only objects in the range -1.5 < V-I < 1.5.  This large query ultimately returns more than 700,000 objects and takes about a minute to complete."
```

Comment:
Might be a good idea to bold-face the "takes about a minute to complete" and/or put it on its own line.  Novice readers might not read that and think that something is wrong here since it's the first "long-running" cell.

Reply:
done